<!-- <script>
    // const value = false;

    function proms(a,b){
         return new Promise(function(resolve, reject){
       console.log("Fetching the data");
            var c = a / b;
       setTimeout(()=> {
        if(c){
            resolve(`Your answer is ${c}`);
        }else{
            reject(`Your answer can not be found`);
        }
       }, 1000);
        
    });
        
}


proms(0,1)
.then((result) => {
    console.log(result);}) 
.catch((error) => {
        console.log(error);}); 
     
     
</script> -->


<!-- <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
<script>
    
    function proms(){
         return new Promise(function(resolve, reject){
       console.log("Fetching the data");
            
       setTimeout(()=> {
        $.get("https://jsonplaceholder.typicode.com/posts " , function(data) {
            resolve(data);
        }).fail(err => {
            reject("Fail to load JSON");
        })
       }, 2000);
        
    });
        
}
proms()
.then((result) => {
    console.log(result);}) 
.catch((error) => {
        console.log(error);}); 
</script> -->

<!-- <script>
let p1 = new Promise((resolve,reject) => {
    setTimeout(() => {
        console.log('This first promise has resolved');
       resolve(10);
    }, 1*1000);
});

let p2 = new Promise((resolve , reject) => {
    setTimeout(() => {
        console.log('This Second promise has Failed');
        reject("Failed");
    }, 2*1000);
});

let p3 = new Promise((resolve,reject) => {
    setTimeout(() => {
        console.log('This Third promise has resolved');
        resolve(20);
    }, 3*1000);
});

var total = 0;
Promise.all([p1,p2,p3]).then((result) => {
    for(var i in result){
        total += result[i];
    }
    
    // console.log(result);
    console.log(`Total: ${total}`);
    console.log(`Result: ${result}`);
}).catch((errors) => {
    console.log(`Error: ${errors}`); 
})
</script> -->

<script>
    let promiseCall = function(returnData, message){
    return function(resolve, reject){
        setTimeout(() => {
       console.log(`The ${message} promise has resolved`);
       resolve(returnData); 
    }, returnData * 100);
    }
};

let p1 = new Promise(promiseCall(10,"First"));
let p2 = new Promise(promiseCall(20,"Second"));
let p3 = new Promise(promiseCall(30,"Third"));
let p4 = new Promise(function(resolve, reject){
    reject("The 4th Promise has rejected.");
});

var total = 0;
Promise.all([p1,p2,p3]).then((result) => {
    for(var i in result){
        total += result[i]}
        console.log(`Result: ${result}`);
        console.log(`Total: ${total}`);
    
}).catch((error) => {
    console.log(`Error : ${error}`);
}); 
</script>